include/RemotePluginClient.h

```diff
@@ -265,6 +244,14 @@ bool RemotePluginClient::processMessage( const message & _m )
 		case IdUndefined:
 			return false;
 
+		case IdSyncKey:
+			m_vstSyncData.attach(_m.getString(0));
```

Comment:
The same story with previous PR: how would failures be handled here? I'm asking because RemoteZynAddSubFx should still work in case of failures.

Reply:
Shared memory has to be functional on the system for the remote plugins to be useful, since it is used to transfer audio to (and from) LMMS. The shared memory here is handled using the exact same mechanism as that for audio - generate a random UUID, use it to create a shared memory object, send the key to the remote plugin, and use it to attach the shared memory object. I don't see why we would expect this line to potentially fail, but still expect the audio buffer to attach reliably. On Windows, we even use shared memory for sending the IPC messages.

My inclination is that if any shared memory fails to attach, we treat that as a hard error and kill the remote plugin. Here I do that by letting the exception escape; this is admittedly not very elegant, and maybe a log message followed by a call to `exit` would be better, but it's intended behaviour. All other shared memory is essential, and I don't think it's worth complicating the code to allow it to hobble on in the rare case that a single piece of shared memory fails to attach. (Currently, we do try to continue if the audio buffer can't be attached - I intend to change this too, but it's out of scope here.)

Reply:
I'm okay with the decision, but as you said, proper logging and/or cleanup on failure will be better. It's definitely good to handle this in this PR, but you may do that in another PR if you want.